test_scripts/selenium/queryservice-ui/queryservice-ui.page.js

```diff
@@ -17,6 +17,17 @@ class QueryServiceUI extends Page {
 		this.submitBtn.click();
 	}
 
+	resultIncludes( prop, value ) {
+		const text = this.resultTable.getText();
+		if ( !value ) {
+			return text.includes( prop );
+		}
+
+		const regexp = new RegExp( '(' + prop + ')(\\s+)(' + value + ')' );
+		const matches = text.match( regexp );
+		return matches !== null && matches.includes( prop ) && matches.includes( value );
```

Comment:
The additional `String.includes` is redundant here: If the regexp matches, both `prop` and `value` are already part of `text`.

Reply:
It's not a string that comes back from the text.match but an array with all the groups that matched. So the prop could match but the value might not be there.

Reply:
You are right about the array part, but still, matches will be null if only part of the string matched.

```
"foo ba".match(new RegExp( '(foo)(\\s+)(bar)' ));
null
"fo bar".match(new RegExp( '(foo)(\\s+)(bar)' ));
null
"foo bar".match(new RegExp( '(foo)(\\s+)(bar)' ));
Array(4) [ "foo bar", "foo", " ", "bar" ]
```

Reply:
My bad, should have confirmed that. will remove it!